frontend/src/metabase/lib/data_grid.js

```diff
@@ -420,7 +420,7 @@ function addSubtotal(
         },
       ]
     : [];
-  if (item.isCollapsed) {
+  if (item.isCollapsed && hasSubtotal) {
```

Comment:
This is a delightfully simple one-line fix!


But one thing that feels a little weird is how when you collapse a whole column, suddenly the output is not uniform, in that the single items don't look like the totals.

![Screen Shot 2023-03-17 at 9 49 45 AM](https://user-images.githubusercontent.com/30528226/225958661-725dc865-197c-475f-b86c-776fbe2180c9.png)

What if instead we just allow subtotals for single-item groups?

![Screen Shot 2023-03-17 at 10 16 56 AM](https://user-images.githubusercontent.com/30528226/225960890-9badf242-1763-491e-a647-a0e0a7807b12.png)


line 433, below would be:

```
shouldShowSubtotal: true,
```

What do you think?

Reply:
We could do something like that. In this case, I got some feedback from product and this was what they were expecting (I'll send you the loom), but I think the general idea was that should should just act like collapsing the existing subtotals rahter than having subtotal cells generated when they weren't present before

Reply:
maybe we can have our cake and eat it too? With this change we can show the subtotal when the column gets collapsed, and not show subtotals in expanded mode for single-item groups?

```
            shouldShowSubtotal: child.children.length > 1 || child.isCollapsed,
```

![cakeToo](https://user-images.githubusercontent.com/30528226/225969575-5e4b4218-afa6-4e53-9054-da009ca68d40.gif)


@mazameli what do you think?

Reply:
I like the behavior in Ryan's gif here, though I'm befuddled by why the Bug and Steel rows say (7,2) and (4,1) instead of (2,2) and (1,1) — shouldn't they have the same values since there's only one row in each of those groups?

Reply:
the third column is pokemon generation (there are 2 bug pokemon in gen 7), which is nonsensical to total. we're just totaling count

Reply:
I'll make the change and adjust the tests